README
Dist::Zilla::Role::Version::Sanitize



INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-Role-Version-Sanitize

0.002004 2017-03-03T09:09:45Z 2485d2c

0.002003 2017-03-03T08:46:07Z 215ad08
 [Bugfix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Role-Version-Sanitize/issues/2

 [Dependencies::Stats]
 - Dependencies changed since 0.002002, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑1)
 - test: ↓1

0.002002 2016-06-06T22:54:09Z a3dac06
 - Add VERSION to metaconfig.

 [Dependencies::Stats]
 - Dependencies changed since 0.002001, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +16 ↑3 -3 (recommends: +1, suggests: ↑2)
 - runtime: ↓1 -1
 - test: +1 (recommends: +1 ↑1 ↓1)

0.002001 2014-08-15T21:42:46Z
 [00 Trivial]
 - CPANDAY!
 - no code changes.
 - metadata updated.
 - tests augmented.

 [Dependencies::Stats]
 - Dependencies changed since 0.002000, see misc/*.deps* for details
 - configure: ↓1
 - develop: +2 (suggests: ↑1)
 - test: +1 -2 (recommends: ↑1 -1)

 [Metadata]
 - Moved to kentnl org.

 [Tests]
 - Test::ReportPrereqs updated.

0.002000 2014-07-13T21:03:31Z
 [00 Major]
 - Some changes in packaging that might break downstreams that work by hand.
 - Misc other packaging cleanups

 [01 Tooling]
 - Switched to ExtUtils::MakeMaker ( https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f )
 - Toolchain Dependencies softened to recommendations instead of requirements

 [Dependencies::Stats]
 - Dependencies changed since 0.001002, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +6 -8 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +3 ↓1 (recommends: +4)

0.001002 2013-12-10T01:20:15Z
 - Only debug message about normalisation when normalisation is applied

0.001001 2013-12-01T19:26:20Z
 - Document attributes.

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - develop: +1 ↑1 -1

0.001000 2013-12-01T10:03:15Z
 - First version.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.019.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4205"
  },
  "configure_requires" => {
    "Module::Build" => "0.4205"
  },
  "dist_abstract" => "Sanitize a version from a plugin",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Role-Version-Sanitize",
  "dist_version" => "0.001003",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Role::Version::Sanitize",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Moose::Role" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
    "ExtUtils::MakeMaker" => 0,
    "File::Copy::Recursive" => 0,
    "File::ShareDir" => 0,
    "File::Spec::Functions" => 0,
    "List::Util" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
  "ExtUtils::MakeMaker" => 0,
  "File::Copy::Recursive" => 0,
  "File::ShareDir" => 0,
  "File::Spec::Functions" => 0,
  "List::Util" => 0,
  "Module::Build" => "0.4205",
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Sanitize a version from a plugin",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Role-Version-Sanitize",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Role::Version::Sanitize",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Moose::Role" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
    "ExtUtils::MakeMaker" => 0,
    "File::Copy::Recursive" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.002004",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
  "ExtUtils::MakeMaker" => 0,
  "File::Copy::Recursive" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "Moose::Role" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
